universal-login-sdk/lib/api/DeployedWallet.ts

```diff
@@ -39,34 +41,35 @@ export class DeployedWallet implements ApplicationWallet {
     };
   }
 
-  async addKey(publicKey: string, transactionDetails: Partial<Message>): Promise<Execution> {
-    return this.sdk.addKey(this.contractAddress, publicKey, this.privateKey, transactionDetails);
+  async addKey(publicKey: string, executionOptions: ExecutionOptions): Promise<Execution> {
+    return this.sdk.addKey(this.contractAddress, publicKey, this.privateKey, {gasLimit: DEFAULT_GAS_LIMIT, ...executionOptions});
   }
 
-  async addKeys(publicKeys: string[], transactionDetails: Partial<Message>): Promise<Execution> {
-    return this.sdk.addKeys(this.contractAddress, publicKeys, this.privateKey, transactionDetails);
+  async addKeys(publicKeys: string[], executionOptions: ExecutionOptions): Promise<Execution> {
+    return this.sdk.addKeys(this.contractAddress, publicKeys, this.privateKey, {gasLimit: DEFAULT_GAS_LIMIT, ...executionOptions});
   }
 
-  async removeKey(key: string, transactionDetails: Partial<Message>): Promise<Execution> {
-    return this.sdk.removeKey(this.contractAddress, key, this.privateKey, transactionDetails);
+  async removeKey(key: string, executionOptions: ExecutionOptions): Promise<Execution> {
+    return this.sdk.removeKey(this.contractAddress, key, this.privateKey, {gasLimit: DEFAULT_GAS_LIMIT, ...executionOptions});
   }
 
-  async removeCurrentKey(transactionDetails: Partial<Message>): Promise<Execution> {
+  async removeCurrentKey(executionOptions: ExecutionOptions): Promise<Execution> {
     const ownKey = utils.computeAddress(this.privateKey);
-    return this.removeKey(ownKey, transactionDetails);
+    return this.removeKey(ownKey, executionOptions);
   }
 
   async denyRequests() {
     return this.sdk.denyRequests(this.contractAddress, this.privateKey);
   }
 
-  async setRequiredSignatures(requiredSignatures: number, transactionDetails: Partial<Message>): Promise<Execution> {
-    return this.sdk.setRequiredSignatures(this.contractAddress, requiredSignatures, this.privateKey, transactionDetails);
+  async setRequiredSignatures(requiredSignatures: number, executionOptions: ExecutionOptions): Promise<Execution> {
+    return this.sdk.setRequiredSignatures(this.contractAddress, requiredSignatures, this.privateKey, {gasLimit: DEFAULT_GAS_LIMIT, ...executionOptions});
   }
 
   async execute(message: Partial<Message>): Promise<Execution> {
```

Comment:
Shouldn't be here ExecutionOptions?

Reply:
Execute calls sdk.execute function which is more complex than executionOptions e.g. it should get parameter `to`.

Furthermore, execute is not called by any other function from deployedWallet.